plutus-tx/src/PlutusTx/AssocMap.hs

```diff
@@ -151,9 +160,12 @@ instance (Eq k, Semigroup v) => Monoid (Map k v) where
 instance (Pretty k, Pretty v) => Pretty (Map k v) where
   pretty (Map mp) = pretty mp
 
-{-# INLINEABLE fromList #-}
-fromList :: [(k, v)] -> Map k v
-fromList = Map
+{-# INLINEABLE unsafeFromList #-}
+-- | Unsafely create a 'Map' from a list of pairs. This should _only_ be applied to lists which
+-- have been checked to not contain duplicate pairs or duplicate keys, otherwise the 'Map' invariant
+-- will be broken. As usual, the "keys" are considered to be the first element of the pair.
+unsafeFromList :: [(k, v)] -> Map k v
+unsafeFromList = Map
 
 {-# INLINEABLE fromListSafe #-}
```

Comment:
If the other one is called `unsafeFromList`, it's better to call this one `safeFromList`.

Reply:
I think there's some confusion here. The other one is called `safeFromList` (and was already called that), and it is safe because it inserts new entries and removes duplicates. The one I renamed to `unsafeFromList` used to be called `fromList` and it is the unsafe one.